### PR TITLE
Final Python 3.7 -> 3.10 updates

### PR DESCRIPTION
**Pull request by spham-amzn**
## What does this PR do?

House-keeping updates for references to the previous version of Python (3.7)

* Update references to Python 3.7(.12) to 3.10(.5) in scripts, comments, etc
* Also Remove unnecessary libffi installation for ubuntu 20.04

## How was this PR tested?

Most of the changes are in comments, messages only, not any direct code. The only scripting code that is affected is used to configure new build nodes.

### CODE REVIEW

**Review by rgba16f — APPROVED** [review]

**Review by brianherrera — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by amzn-changml — APPROVED** [review]

File: scripts/build/build_node/Platform/Windows/install_python.ps1 [hunk 1]
```diff
@@ -6,7 +6,7 @@ SPDX-License-Identifier: Apache-2.0 OR MIT
 #>
 
 choco install -y python2 --version=2.7.15
```
Comment (lmbr-pip) [inline scripts/build/build_node/Platform/Windows/install_python.ps1:8]: What do we have that still uses 2.7? Py 2.7 reached end of life January 1st, 2020

**Review by lmbr-pip — APPROVED** [review]
Thanks for doing this cleanup

File: scripts/build/build_node/Platform/Windows/install_python.ps1 (hunk 1 above)
Comment (spham-amzn) [reply scripts/build/build_node/Platform/Windows/install_python.ps1:8]: I'm not sure, @amzn-changml might know why we need this on the Windows build nodes

Comment (amzn-changml) [reply scripts/build/build_node/Platform/Windows/install_python.ps1:8]: We don't need it anymore, it was a legacy of pre-build scripts that were originally in py2. That should be removed

Comment (spham-amzn) [reply scripts/build/build_node/Platform/Windows/install_python.ps1:8]: Ok, i'll remove it as part of this PR as well

Comment (amzn-changml) [reply scripts/build/build_node/Platform/Windows/install_python.ps1:8]: Thanks! If it's ok, can you also update the env var path in line 14 to reflect py 3.10's path?

**Review by amzn-changml — APPROVED** [review]

**Pull request merged**